src/ORM/Association/Loader/SelectLoader.php

```diff
@@ -77,20 +153,17 @@ protected function _defaultOptions()
      */
     protected function _buildQuery($options)
     {
-        $target = $this->target();
-        $alias = $target->alias();
+        $alias = $this->targetAlias;
         $key = $this->_linkField($options);
         $filter = $options['keys'];
-        $useSubquery = $options['strategy'] === $this::STRATEGY_SUBQUERY;
+        $useSubquery = $options['strategy'] === Association::STRATEGY_SUBQUERY;
+        $finder = $this->finder;
 
-        $finder = isset($options['finder']) ? $options['finder'] : $this->finder();
-        list($finder, $opts) = $this->_extractFinder($finder);
         if (!isset($options['fields'])) {
             $options['fields'] = [];
         }
 
-        $fetchQuery = $this
-            ->find($finder, $opts)
+        $fetchQuery = $finder()
```

Comment:
Kind of nice that we get to reuse the `find()` logic in `Association::find()` now.